src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -20,10 +21,9 @@ def initialize(message, cause = nil)
   end
 end
 
-class Puppet::Server::HttpClient
+class Puppet::Server::HttpClient < Puppet::HTTP::Client
 
   OPTION_DEFAULTS = {
-      :use_ssl => true,
       :verify => nil,
       :redirect_limit => 10,
```

Comment:
The new client accepts keyword arguments for `:pool`, `:ssl_context`, `:system_ssl_context`, :redirect_limit` and `:retry_limit`. For now I think only the latter two make sense for puppetserver. The `:redirect_limit` can also be overridden on a per-request basis. So 👍  to dropping `:use_ssl` and `:verify`.

Reply:
Okay I'll update this to make sure that works here.

Reply:
Currently those options are not honored or passed to the Java implementation when this class is created. The client itself is created lazily with an entirely different set of methods, and the `ClientOptions` class from the clojure HTTP client currently does not support either of those options. I'll have to dig some more.

Reply:
Notes for myself: The retry limit can be configured by setting a `RetryHandler` on the client obejct ([example](https://memorynotfound.com/apache-httpclient-httprequestretryhandler-example/)).

The redirect limit is apparently configurable at least on the client, though I haven't been able to wade through the layers of Java abstraction to figure out exactly how yet.

Reply:
I looked into how to support configuring `redirect_limit` in our Clojure client, and the APIs we use to build the Java client class are the wrong level of abstraction to set it. Looks like the default redirect_limit is 100 (which is much higher than Puppet’s default, but at least it is limited). Considering it wasn’t configurable before, we are going to leave it unconfigurable now.

It also appears that the version of the HttpAsyncClient we’re using doesn’t support retries at all. So I think configuring that isn’t going to happen. They added it in the next major version, which has a totally different API.